### PR TITLE
fix: [PL-23324]: fix the issue due to which branch sync is stuck

### PR DESCRIPTION
**Pull request by bhavya181**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)

### CODE REVIEW

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)* status: 8 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java, line 25 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxF9GMd7SV286JXIzwe:-MxF9GMd7SV286JXIzwf:bvhaj5g) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java#L25)):*
> ```Java
> @Slf4j
> @OwnedBy(PL)
> public class MarkStuckBranchesStatusToUnsyncMigration implements NGMigration {
> ```

UpdateStuckGitBranchesStatusToUnsynced

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)* status: 8 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java, line 178 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxFDpUMAKYcKb4ull3W:-MxFDpUMAKYcKb4ull3X:b-erig8i) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java#L178)):*
> ```Java
>   protected String getCommitMessage(YamlGitConfigDTO yamlGitConfig, String commitId, String accountIdentifier) {
>     if (commitId == null) {
>       return GitSyncErrorUtils.EMPTY_STR;
> ```

empty string means we got some commit message but for null commitId we didnt got any message and i think returning empty string will be wrong

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java [hunk 1]
```diff
@@ -174,7 +175,7 @@ private String getCommitId(List<GitFileChangeDTO> harnessFilesOfBranch) {
   @VisibleForTesting
   protected String getCommitMessage(YamlGitConfigDTO yamlGitConfig, String commitId, String accountIdentifier) {
     if (commitId == null) {
-      return null;
+      return GitSyncErrorUtils.EMPTY_STR;
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java:178]: why -- looks like a hack

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)* status: 8 file(s) to review. 3 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java, line 178 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIOfhTCQaHqcrjeqdZ:-MxIOfhU5nC_iZVajI3H:b-p3ejvi) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java#L178)):*
> ```Java
>   protected String getCommitMessage(YamlGitConfigDTO yamlGitConfig, String commitId, String accountIdentifier) {
>     if (commitId == null) {
>       return GitSyncErrorUtils.EMPTY_STR;
> ```

null was looking correct
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/impl/BranchPushEventYamlChangeSetHandler.java, line 159 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIOmmTAcOrunTiYffX:-MxIOmmU2u9DeI3q16QK:bmhxkko) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/core/impl/BranchPushEventYamlChangeSetHandler.java#L159)):*
> ```Java
>         log.error("Error while processing branch push event {}", yamlChangeSetDTO, ex);
>         String gitConnectivityErrorMessage = GitConnectivityExceptionHelper.getErrorMessage(ex);
>         if (!isEmpty(gitConnectivityErrorMessage)) {
> ```

isNotEmpty can also be used
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java, line 34 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIPJ39DpPVYa4UVkX4:-MxIPJ39DpPVYa4UVkX5:b-76x2o5) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java#L34)):*
> ```Java
>     log.info("Starting migration to change the status of stuck branches");
>     try {
>       mongoTemplate.updateMulti(query(criteria), update, GitBranch.class);
> ```

Can we also log how many entities were updated ? Maybe if we can log their ids then it will be helpful
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java, line 38 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIPCIT5TZtWSZ5tXuT:-MxIPCIT5TZtWSZ5tXuU:b-vqhy7n) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java#L38)):*
> ```Java
>       log.error("Error occurred while migrating stuck branches", ex);
>     }
>   }
> ```

Add a log line that migration completed

**Review by ankit-harness — DISMISSED** [review]
:lgtm:

> Reviewed 2 of 8 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157#-:-MxJCM9kAWjwVprTZORX:bnfp4nl)* status: 6 file(s) to review. 2 change requests. 1 of 2 approvals obtained.

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)* status: 6 file(s) to review. 2 change requests. 7. unresolved discussions. 1 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java, line 178 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxEgk4oPuC9WGrfkFoY-r1-178:-MxJC_EK67NkPuqolTHF:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java#L178)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

why -- looks like a hack
</blockquote></details>

Done.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java, line 178 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIOfhTCQaHqcrjeqdZ:-MxJC_Z_DCQVX0ZlTA-R:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java#L178)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

null was looking correct
</blockquote></details>

Done.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java, line 178 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxFDpUMAKYcKb4ull3W:-MxJC_k18-6avT7Ip12O:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/GitBranchSyncServiceImpl.java#L178)):*
<details><summary><i>Previously, rathodmeetsatish (Meet Rathod) wrote…</i></summary><blockquote>

empty string means we got some commit message but for null commitId we didnt got any message and i think returning empty string will be wrong
</blockquote></details>

Done.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/impl/BranchPushEventYamlChangeSetHandler.java, line 159 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIOmmTAcOrunTiYffX:-MxJCaEy2jBBPkJAwDb1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/core/impl/BranchPushEventYamlChangeSetHandler.java#L159)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

isNotEmpty can also be used
</blockquote></details>

Done.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java, line 25 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxF9GMd7SV286JXIzwe:-MxJCad71iPqeXvQLhhq:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java#L25)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

UpdateStuckGitBranchesStatusToUnsynced
</blockquote></details>

Done.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java, line 34 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIPJ39DpPVYa4UVkX4:-MxJCcd31AeYqfH7mj16:bmhfsd0) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java#L34)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Can we also log how many entities were updated ? Maybe if we can log their ids then it will be helpful
</blockquote></details>

We can't show Ids, but the number can be shown
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java, line 38 at r1](https://reviewable.io/reviews/harness/harness-core/31157#-MxIPCIT5TZtWSZ5tXuT:-MxJCgfSA9eBZJqFyvz3:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/77187e67fbc2032e69447fafd94ee74a3e992106/136-git-sync-manager/src/main/java/io/harness/gitsync/migration/MarkStuckBranchesStatusToUnsyncMigration.java#L38)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Add a log line that migration completed
</blockquote></details>

Done.

**Review by DeepakPatankar — DISMISSED** [review]
> Reviewed 5 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)* status: 1 file(s) to review. 1 change requests. 3. unresolved discussions.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)* status: 4 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained.

**Comment by bhavya181** [comment]
trigger messagecheck

**Review by rathodmeetsatish — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157)* status: 4 file(s) to review. 2. unresolved discussions.

**Comment by bhavya181** [comment]
trigger messagecheck

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157#-:-MxKEky1-VsgpsMxqlJ5:bnfp4nl)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> Reviewed 5 of 6 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31157#-:-MxLMmIS4QoQUBuWKpEz:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**